src-electron/db/query-config.js

```diff
@@ -222,14 +223,16 @@ async function insertOrUpdateAttributeState(
     staticAttribute.defaultValue == 0
   ) {
     let featureMapDefaultValue = staticAttribute.defaultValue
-    let mandatoryFeaturesOnEndpointTypeAndCluster =
+    let featuresOnEndpointTypeAndCluster =
       await queryDeviceType.selectDeviceTypeFeaturesByEndpointTypeIdAndClusterId(
         db,
         endpointTypeId,
         clusterRef
       )
-    let featureMapBitsToBeEnabled =
-      mandatoryFeaturesOnEndpointTypeAndCluster.map((f) => f.featureBit)
+    // only set featureMap bit to 1 for mandatory features
+    let featureMapBitsToBeEnabled = featuresOnEndpointTypeAndCluster
+      .filter((f) => f.conformance == 'M')
```

Comment:
Hardcoding "M" seems suspicious. This is:
   1.) Matter specific. Zigbee knows nothing about what on earth "M" is, so putting "M" anywhere in the code, is a data point from Matter spec, making way into Zap codebase. And this is in `db` package, which is definitely not Matter specific.

Reply:
The entire code section of setting featureMap bits here is too Matter-specific and should be restructured in the future. I created an issue on this: #1488